### PR TITLE
Premint - can execute premint on legacy version of contracts

### PR DESCRIPTION
**Pull request by oveddan**
If an 1155 contract was created with the old function delegateSetupNewToken() - the upgraded premint executor could not execute that function cause it expected the newer signature with different arguments.  This PR will have the premint executor check if the contract supports the new function, and if it doesn't, then it calls the old function

### CODE REVIEW

**Comment by oveddan** [comment]
Current dependencies on/for this PR:
* main
  * **PR #278** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/278" target="_blank">Graphite</a> 
    * **PR #297** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/297" target="_blank">Graphite</a> 
    * **PR #290** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/290" target="_blank">Graphite</a> 
      * **PR #291** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/291" target="_blank">Graphite</a> 
        * **PR #262** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/262" target="_blank">Graphite</a> 
          * **PR #294** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/294" target="_blank">Graphite</a> 
            * **PR #295** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/295" target="_blank">Graphite</a> 
              * **PR #289** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/289" target="_blank">Graphite</a> 
                * **PR #299** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/299" target="_blank">Graphite</a>  👈
          * **PR #273** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/273" target="_blank">Graphite</a> 
            * **PR #288** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/288" target="_blank">Graphite</a> 
              * **PR #284** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/284" target="_blank">Graphite</a> 
            * **PR #280** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/280" target="_blank">Graphite</a> 
    * **PR #283** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/283" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ourzora/zora-protocol/299?utm_source=stack-comment).

File: packages/1155-contracts/src/delegation/ZoraCreator1155Attribution.sol [hunk 1]
```diff
@@ -411,6 +411,8 @@ library DelegatedTokenCreation {
             );
 
             (params, tokenSetupActions) = _recoverDelegatedTokenSetup(premintConfig, newTokenId);
+        } else {
```
Comment (IsabellaSmallcombe) [inline packages/1155-contracts/src/delegation/ZoraCreator1155Attribution.sol:414]: might be cleaner to remove the extra else and just revert

File: packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImplLib.sol [hunk 2]
```diff
@@ -60,6 +65,21 @@ library ZoraCreator1155PremintExecutorImplLib {
...
 
+    function legacySetupNewToken(address contractAddress, bytes memory encodedPremintConfig, bytes calldata signature) private returns (uint256) {
+        // for use when the erc1155 contract does not support the new delegateSetupNewToken interface, where it expects
+        // a PremintConfig as an argument.
+
+        // decode the PremintConfig from the encoded bytes.
+        PremintConfig memory premintConfig = abi.decode(encodedPremintConfig, (PremintConfig));
+
+        // call the legacy version of the delegateSetupNewToken function.
+        return ILegacyZoraCreator1155DelegatedMinter(contractAddress).delegateSetupNewToken(premintConfig, signature, msg.sender);
```
Comment (iainnash) [inline packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImplLib.sol:76]: What happens for contracts deployed pre-premint? does this just revert?

**Review by iainnash — DISMISSED** [review]

File: packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImplLib.sol (hunk 2 above)
Comment (oveddan) [reply packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImplLib.sol:76]: yes they'd revert

**Review by kulkarohan — APPROVED** [review]

**Review by IsabellaSmallcombe — DISMISSED** [review]

**Comment by oveddan** [comment]
### Merge activity

* **Nov 15, 2:48 PM**: @oveddan started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/ourzora/zora-protocol/299).

**Pull request merged**